scripts/ddtest

```diff
@@ -10,7 +10,7 @@ then
 fi
 
 # retry docker pull if fails
-for i in {1..3}; do docker-compose pull -q testrunner && break || sleep 3; done
```

Comment:
Do we know for a fact that it's safe to sunset `docker-compose` across the invocations of this script?  I ask out of ignorance, since I don't know the coverage for when `compose` was integrated into the `docker` CLI.

If we _don't_ know, may I propose something like the following?
```
get_docker_compose_command() {
    if docker compose version &> /dev/null; then
        echo "docker compose"
    elif docker-compose version &> /dev/null; then
        echo "docker-compose"
    else
        echo "Error: Neither 'docker compose' nor 'docker-compose' is available." >&2
        exit 1
    fi
}
```

I _think_ both `docker-compose` and `docker compose` provide a `version` parameter, which should make it straightforward to implement a fallback.  Something like

```
DOCKER_COMPOSE=$(get_docker_compose_command)
```

So then lines like this would become
```
for i in {1..3}; do $DOCKER_COMPOSE pull -q testrunner && break || sleep 3; done
```

(I wrote all of this out in a GH comment, so it is totally off-the-cuff and untested; just trying to help think of consequences)

Reply:
Good call out, thanks for the suggestion. I like this safer approach